### ISSUE TITLE
Advice on how to install GhostFaceNet

### ISSUE THREAD

**Issue by tomwieden** [serengil/deepface#1155]
All models except GhostFaceNet can be installed via pip.

Any hint on how to easily install GhostFaceNet?

**Comment by serengil** [comment]
What is the problem?

**Comment by serengil** [comment]
Oh i see, you do not have to do something special if you install the latest deepface.

Just run this

result = DeepFace.verify(img1_path = "img1.jpg",  img2_path = "img2.jpg",  model_name = "GhostFaceNet")

This will download its pre-trained weights in the background automatically.

**Comment by tomwieden** [comment]
Many thanks for the hint!

I can confirm your solution...

    pip install --upgrade deepface

**Comment by tomwieden** [comment]
wait a minute... this piece of code...

```
resp_objs = DeepFace.represent(
                    img_path=file,
                    model_name="GhostFaceNet",
                    enforce_detection=False,
                    detector_backend="retinaface",
                    align=True,
                    expand_percentage=0,
                    normalization="base",
                )
```

still throws...

    ImportError('Keras cannot be imported. Check that it is installed.')

**Comment by tomwieden** [comment]
This is a bit strange because if I do

    sh-5.2$ pip install keras

it shows 

```
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: keras in /home/nuc/.var/app/com.vscodium.codium/data/python/lib/python3.11/site-packages (2.15.0)
```

Environment: Debian testing, VSCodium

**Comment by tomwieden** [comment]
BUT the same code works if I combine GhostFaceNet and yolov8

```
resp_objs = DeepFace.represent(
                    img_path=file,
                    model_name="GhostFaceNet",
                    enforce_detection=False,
                    detector_backend="yolov8",
                    align=True,
                    expand_percentage=0,
                    normalization="base",
                )
```

**Comment by serengil** [comment]
What is your tf version? Downgrade it to 2.15 if it is 2.16.

Keras cannot be imported message is not coming from deepface.

**Comment by tomwieden** [comment]
It is not GhostFaceNet that cause the trouble. Instead it is retinaface in combination with some models.

Many thanks for you fast replies and you patience!

I will try a bit more and then report.

And yes - I messed around with the upgrade and the "eager" parameter and it might be a mess now.

**Comment by tomwieden** [comment]
After executing

    pip install --upgrade tf-keras==2.15 tensorflow==2.15

the combination of GhostFaceNet and retinaface does work now.

Just for interest.... I included tf-keras==2.15 because the downgrade of tensorflow to 2.15 showed an

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tf-keras 2.16.0 requires tensorflow<2.17,>=2.16, but you have tensorflow 2.15.0 which is incompatible.
```

Thanks again for your time and support!

**Comment by serengil** [comment]
Happy to hear
